src/Sulu/Bundle/AudienceTargetingBundle/Tests/Functional/Controller/TargetGroupControllerTest.php

```diff
@@ -0,0 +1,245 @@
+
+        $client->request('PUT', self::BASE_URL . '/' . $this->targetGroup->getId(), $sampleData);
+
+        $this->assertHttpStatusCode(200, $client->getResponse());
+        $response = json_decode($client->getResponse()->getContent(), true);
+
+        $this->assertEquals($sampleData['title'], $response['title']);
+        $this->assertEquals($sampleData['description'], $response['description']);
+        $this->assertEquals($sampleData['priority'], $response['priority']);
+        $this->assertEquals($sampleData['active'], $response['active']);
+        $this->assertCount(count($sampleData['webspaces']), $response['webspaces']);
+        $this->assertEquals($sampleData['webspaces'][0]['webspaceKey'], $response['webspaces'][0]['webspaceKey']);
+
+        $this->assertNotNull($this->getTargetGroupRepository()->find($response['id']));
+    }
+
+    /**
+     * Test deleting a target group over api.
+     */
+    public function testSingleDelete()
+    {
+        $client = $this->createAuthenticatedClient();
+        $client->request('DELETE', self::BASE_URL . '/' . $this->targetGroup->getId());
+
+        $response = $client->getResponse();
+        $this->assertHttpStatusCode(204, $response);
+
+        $this->getEntityManager()->clear();
+
+        $targetGroup = $this->getTargetGroupRepository()->find($this->targetGroup->getId());
+
+        $this->assertNull($targetGroup);
+    }
+
+    /**
+     * Test deleting multiple target groups over api.
+     */
+    public function testMultipleDelete()
+    {
+        $client = $this->createAuthenticatedClient();
+        $client->request(
+            'DELETE',
+            self::BASE_URL . '?ids=' . implode(',', [$this->targetGroup->getId(), $this->targetGroup2->getId()])
+        );
+
+        $response = $client->getResponse();
+        $this->assertHttpStatusCode(204, $response);
+
+        $this->getEntityManager()->clear();
+
+        $targetGroup = $this->getTargetGroupRepository()->find($this->targetGroup->getId());
+        $targetGroup2 = $this->getTargetGroupRepository()->find($this->targetGroup2->getId());
+
+        $this->assertNull($targetGroup);
+        $this->assertNull($targetGroup2);
+    }
+
+    /**
+     * Returns sample data for target group creation as array.
+     *
+     * @return array
+     */
+    private function getSampleData()
+    {
+        return [
+            'title' => 'Target Group Title',
+            'description' => 'Target group description number 1',
+            'priority' => 3,
+            'active' => true,
+            'webspaces' => [
+                [
+                    'webspaceKey' => 'my-webspace-1',
+                ],
+                [
+                    'webspaceKey' => 'my-webspace-2',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Returns alternative sample data for target group changes.
+     *
+     * @return array
+     */
+    private function getAlternativeSampleData()
```

Comment:
Instead of creating a single function for every dataset I would declare them in every test separately. Otherwise you are coupling them very hard, and if you want to add something here because you are adding a more special test case it might break other tests as well...